Initial commit for sdk

Hemangi-Chauhan opened the pull request.

Hemangi-Chauhan committed:
Initial commit for sdk

Hemangi-Chauhan committed:
Initial commit for sdk

dftrakesh commented on pom.xml:
```diff
@@ -0,0 +1,168 @@
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.dft</groupId>
```
it should be fixed io.github.dftrakesh

Hemangi-Chauhan replied:
Change done.

dftrakesh commented on pom.xml:
```diff
@@ -0,0 +1,168 @@
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>org.dft</groupId>
+    <artifactId>VeracrossSDK</artifactId>
```
should be veracross-sdk

Hemangi-Chauhan replied:
Changed to veracross-sdk.

dftrakesh commented on pom.xml:
```diff
@@ -0,0 +1,168 @@
+            <artifactId>jackson-databind</artifactId>
+            <version>2.13.4</version>
+        </dependency>
+        <dependency>
```
Change dependency to

<dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <version>1.18.24</version>
        </dependency>

dftrakesh replied:
Remove  <scope>compile</scope>

Hemangi-Chauhan replied:
> Remove compile

Removed

dftrakesh commented on src/main/java/com/dft/veracross/VeracrossParents.java:
```diff
@@ -0,0 +1,44 @@
+import java.net.http.HttpResponse;
+import java.util.HashMap;
+
+public class VeracrossParents extends VeracrossSDK {
```
Rename class name to Parents

Hemangi-Chauhan replied:
> Rename class name to Parents

Name changed to Parent.

dftrakesh commented on src/main/java/com/dft/veracross/VeracrossParents.java:
```diff
@@ -0,0 +1,44 @@
+
+public class VeracrossParents extends VeracrossSDK {
+
+    private static final String FORWARD_SLASH_CHARACTER = "/";
```
Move this constant code to VeracrossSDK

Hemangi-Chauhan replied:
> Move this constant code to VeracrossSDK

Moved.

dftrakesh commented on src/main/java/com/dft/veracross/VeracrossParents.java:
```diff
@@ -0,0 +1,44 @@
+        URI uri = baseUrl(PARENTS_ENDPOINT.concat(FORWARD_SLASH_CHARACTER)
+                .concat(id.toString()));
+
+        System.out.println("uri: " + uri);
```
Remove System out use @Log4j2

Hemangi-Chauhan replied:
> Remove System out use @log4j2

Removed unnecessary code from everywhere.

dftrakesh commented on src/main/java/com/dft/veracross/VeracrossSDK.java:
```diff
@@ -0,0 +1,103 @@
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+@AllArgsConstructor
```
Change it to ReuqiredArgsConstrctor

dftrakesh commented on src/main/java/com/dft/veracross/VeracrossSDK.java:
```diff
@@ -0,0 +1,103 @@
+@Builder(builderMethodName = "newBuilder", toBuilder = true)
+public class VeracrossSDK {
+
+    private final VeracrossCredentials credentials;
```
Arrange all fileds with similar type

Hemangi-Chauhan replied:
> Arrange all fileds with similar type

Arranged with similar field.

dftrakesh commented on src/main/java/com/dft/veracross/VeracrossSDK.java:
```diff
@@ -0,0 +1,103 @@
+    int TIME_OUT_DURATION = 60000;
+    private final HttpClient client;
+    private static final String AUTHORIZATION = "Authorization";
+    private static final String BASE_ENDPOINT = "https://api.veracross.com/api-sandbox/v3";
```
By default it should be production URL
Add one constructor to set like isSanbox. If its passed then URL will be changed to Sandbox.

Hemangi-Chauhan replied:
> By default it should be production URL Add one constructor to set like isSanbox. If its passed then URL will be changed to Sandbox.

Now i added sandbox in creadential class, now scool rout pass through credential. so not get issue when it comes to production url.

dftrakesh commented on src/main/java/com/dft/veracross/VeracrossSDK.java:
```diff
@@ -0,0 +1,103 @@
+            pathWithQueryString += "?" + queryString.toString();
+        }
+
+        return new URI(uri.getScheme(), uri.getAuthority(), pathWithQueryString, uri.getFragment());
```
Use URI.create

Hemangi-Chauhan replied:
> 
Changed to URI.create

dftrakesh commented on src/main/java/com/dft/veracross/VeracrossSDK.java:
```diff
@@ -0,0 +1,103 @@
+                .build();
+    }
+
+    @SneakyThrows
```
Remove @SneakyThrows after changing new URI to URI.create 
check next comment.

dftrakesh replied:
Do same for other methods.

Hemangi-Chauhan replied:
Changed to URI.create and removed @SneakyThrows.

dftrakesh commented on src/main/java/com/dft/veracross/VeracrossSDK.java:
```diff
@@ -0,0 +1,103 @@
+                .toString());
+    }
+
+    @SneakyThrows
```
What error it is throwing ?

dftrakesh commented on src/main/java/com/dft/veracross/VeracrossSDK.java:
```diff
@@ -0,0 +1,103 @@
+                                                            HttpResponse.BodyHandler<T> handler,
+                                                            HttpResponse<T> resp, int count) {
+        if (resp.statusCode() == 429 && count < MAX_ATTEMPTS) {
+            Thread.sleep(TIME_OUT_DURATION);
```
https://api-docs.veracross.com/docs/docs/ZG9jOjEyNzQyNjkz-rate-limiting

Implement rate limiting as per documenttaion. Check shipstation SDK for reference.

Hemangi-Chauhan replied:
> https://api-docs.veracross.com/docs/docs/ZG9jOjEyNzQyNjkz-rate-limiting
> 
> Implement rate limiting as per documenttaion. Check shipstation SDK for reference.

Implemented as per documentation

dftrakesh commented on src/main/java/com/dft/veracross/VeracrossSDK.java:
```diff
@@ -0,0 +1,103 @@
+        return CompletableFuture.completedFuture(resp);
+    }
+
+    public VeracrossParents getParentApi() {
```
Remove this method

Hemangi-Chauhan replied:
> Remove this method

Removed.

dftrakesh commented on src/main/java/com/dft/veracross/VeracrossSDK.java:
```diff
@@ -0,0 +1,103 @@
+        return new VeracrossParents(credentials);
+    }
+
+    public VeracrossStudents getStudentApi() {
```
Remove this method

Hemangi-Chauhan replied:
> Remove this method

Removed.

Hemangi-Chauhan committed:
ver-1

Changes groupId and artifactId from pom.xml.
Rename class name Parent and Students from VeracrossParents and VeracrossStudents.
Moved constant code to VeracrossSDK.
Removed println.
Changed RequiredArgsConstructor instead of AllArgsConstructor.
ArrangeD all fileds with similar type in VeracrossSDK.
Used URI.create in VeracrossSDK.
Removed @SneakyThrows annotation.
Made changes of rate limiting.
Removed getParentApi and getStudentApi method.

dftrakesh commented on src/main/java/com/dft/veracross/credentials/VeracrossCredentials.java:
```diff
@@ -0,0 +1,16 @@
+import lombok.NoArgsConstructor;
+
+@Data
+@NoArgsConstructor
```
Why do you need @NoArgsConstructor

Hemangi-Chauhan replied:
> Why do you need @NoArgsConstructor

Sir, i used for testing in test class.

dftrakesh commented on src/main/java/com/dft/veracross/model/parents/Parents.java:
```diff
@@ -0,0 +1,31 @@
+    public Spouse spouse;
+    public Integer gender;
+    public String roles;
+    public Object email1;
```
It should String not object

Hemangi-Chauhan replied:
> It should String not object

Set proper datatype everywhere as per data response

dftrakesh commented on src/main/java/com/dft/veracross/model/parents/parentWrapper/Address.java:
```diff
@@ -0,0 +1,22 @@
+public class Address {
+
+    public String address1;
+    public Object address2;
```
Object ?

dftrakesh commented on src/main/java/com/dft/veracross/model/parents/parentWrapper/Address.java:
```diff
@@ -0,0 +1,22 @@
+
+    public String address1;
+    public Object address2;
+    public Object address3;
```
Object ?

dftrakesh commented on test:
Remove TEST file.

dftrakesh commented on src/main/java/com/dft/veracross/model/students/studentWrapper/StudentInfo.java:
```diff
@@ -0,0 +1,18 @@
+@Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
+@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
```
Do you need this annotation for this class ?

Hemangi-Chauhan replied:
> Do you need this annotation for this class ?

No, i removed.

dftrakesh commented on src/main/java/com/dft/veracross/model/students/studentWrapper/StudentInfo.java:
```diff
@@ -0,0 +1,18 @@
+
+@Data
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonIgnoreProperties(ignoreUnknown = true)
```
Same here

dftrakesh commented on src/main/java/com/dft/veracross/model/students/studentWrapper/StudentData.java:
```diff
@@ -0,0 +1,39 @@
+    public Integer id;
+    public Integer namePrefix;
+    public String firstName;
+    public Object middleName;
```
Provide proper data type please. for all Object

Hemangi-Chauhan replied:
> Provide proper data type please. for all Object

Provided proper datatype in everyclass.

Hemangi-Chauhan committed:
ver-1

Changes groupId and artifactId from pom.xml.
Rename class name Parent and Students from VeracrossParents and VeracrossStudents.
Moved constant code to VeracrossSDK.
Removed println.
Changed RequiredArgsConstructor instead of AllArgsConstructor.
ArrangeD all fileds with similar type in VeracrossSDK.
Used URI.create in VeracrossSDK.
Removed @SneakyThrows annotation.
Made changes of rate limiting.
Removed getParentApi and getStudentApi method.

Hemangi-Chauhan committed:
ver-1

Changes groupId and artifactId from pom.xml.
Rename class name Parent and Students from VeracrossParents and VeracrossStudents.
Moved constant code to VeracrossSDK.
Removed println.
Changed RequiredArgsConstructor instead of AllArgsConstructor.
ArrangeD all fileds with similar type in VeracrossSDK.
Used URI.create in VeracrossSDK.
Removed @SneakyThrows annotation.
Made changes of rate limiting.
Removed getParentApi and getStudentApi method.

Hemangi-Chauhan committed:
ver-1

Changes groupId and artifactId from pom.xml.
Rename class name Parent and Students from VeracrossParents and VeracrossStudents.
Moved constant code to VeracrossSDK.
Removed println.
Changed RequiredArgsConstructor instead of AllArgsConstructor.
ArrangeD all fileds with similar type in VeracrossSDK.
Used URI.create in VeracrossSDK.
Removed @SneakyThrows annotation.
Made changes of rate limiting.
Removed getParentApi and getStudentApi method.
